packages/agreement-platformstate-writer/test/consumerServiceV1.integration.test.ts

```diff
@@ -0,0 +1,1932 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+/* eslint-disable @typescript-eslint/no-floating-promises */
+import { fail } from "assert";
+import {
+  afterAll,
+  afterEach,
+  beforeAll,
+  beforeEach,
+  describe,
+  expect,
+  it,
+  vi,
+} from "vitest";
+import {
+  Agreement,
+  AgreementActivatedV1,
+  AgreementAddedV1,
+  AgreementEventEnvelope,
+  AgreementUpdatedV1,
+  EServiceId,
+  PlatformStatesAgreementEntry,
+  PlatformStatesCatalogEntry,
+  TenantId,
+  TokenGenerationStatesClientPurposeEntry,
+  agreementState,
+  generateId,
+  itemState,
+  makeGSIPKConsumerIdEServiceId,
+  makePlatformStatesAgreementPK,
+  makePlatformStatesEServiceDescriptorPK,
+  makeTokenGenerationStatesClientKidPurposePK,
+} from "pagopa-interop-models";
+import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
+import {
+  getMockTokenStatesClientPurposeEntry,
+  getMockAgreement,
+  buildDynamoDBTables,
+  deleteDynamoDBTables,
+  toAgreementV1,
+  readTokenStateEntriesByConsumerIdEserviceId,
+  getMockAgreementEntry,
+  writeCatalogEntry,
+  writeTokenStateEntry,
+} from "pagopa-interop-commons-test";
+import { readAgreementEntry, writeAgreementEntry } from "../src/utils.js";
+import { handleMessageV1 } from "../src/consumerServiceV1.js";
+import { config, sleep } from "./utils.js";
+
+describe("integration tests V1 events", async () => {
+  if (!config) {
+    fail();
+  }
+  const dynamoDBClient = new DynamoDBClient({
+    endpoint: `http://localhost:${config.tokenGenerationReadModelDbPort}`,
+  });
+  beforeEach(async () => {
+    await buildDynamoDBTables(dynamoDBClient);
+  });
+  afterEach(async () => {
+    await deleteDynamoDBTables(dynamoDBClient);
+  });
+  const mockDate = new Date();
+  beforeAll(() => {
+    vi.useFakeTimers();
+    vi.setSystemTime(mockDate);
+  });
+  afterAll(() => {
+    vi.useRealTimers();
+  });
+
+  describe("AgreementActivated", () => {
+    it("no operation if the entry already exists: incoming has version 1; previous entry has version 2", async () => {
```

Comment:
may we adapt descriptions to other packages?

Reply:
Done